contracts/MultiSigWallet.sol

```diff
@@ -401,99 +384,115 @@ contract MultiSigWallet {
+     * @dev Proposes a transaction with the specified id.
+     *      If transaction was proposed, the function returns successfully.
+     *      Function requires:
+     *          - Transaction was not executed.
      *
-     * @param _transactionId It marks it in proposed state against the wallet
-     *         which has sent the transaction.
+     * @param _transactionId Transaction id to propose.
+     *
+     * @return True in case of success otherwise false.
      */
     function proposeTransaction(
         bytes32 _transactionId
     )
         internal
+        onlyWallet
+        returns (bool)
     {
-        if (isAlreadyProposedTransaction(_transactionId)){
-            return;
+        require(
+            isTransactionExecuted(_transactionId) == false,
+            "Transaction was executed."
+        );
+
+        if (isTransactionProposed(_transactionId)) {
+            return true;
         }
+
+        // Marking the transaction status as proposed.
         confirmations[_transactionId].status = 1;
 
-        emit Proposed(
-            _transactionId,
-            msg.sender
-        );
+        emit Proposed(_transactionId, msg.sender);
+
+        return true;
     }
 
     /**
-     * @notice It is used to send the transaction to confirmation state by the
-     *         wallet who has sent the transaction.
+     * @notice Marks transaction as confirmed by the caller wallet.
+     *
+     * @dev Function requires:
+     *          - The transaction is in proposed state.
+     *          - The transaction was not confirmed by the caller wallet.
      *
-     * @param _transactionId It marks address(this) transaction id as confirmed against
-     *        the wallet which has sent the transaction.
+     * @param _transactionId Transaction id to confirm.
      */
     function confirmTransaction(
         bytes32 _transactionId
     )
         internal
+        onlyWallet
+        returns (bool)
     {
         require(
             confirmations[_transactionId].status == 1,
-            "Please first propose the transaction"
+            "The transaction is not in propose state."
         );
         require(
-            !confirmations[_transactionId].isConfirmedBy[msg.sender],
-            "Transaction is already confirmed by this wallet"
+            isConfirmedBy(_transactionId, msg.sender) == false,
+            "Transaction was confirmed by the caller wallet."
         );
 
         confirmations[_transactionId].isConfirmedBy[msg.sender] = true;
-        emit Confirmed(
-            _transactionId,
-            msg.sender
 
-        );
+        emit Confirmed(_transactionId, msg.sender);
 
-        if (isAlreadyProposedTransaction(_transactionId)) {
-            if (isRequirementAchieved(_transactionId)) {
-                confirmations[_transactionId].status = 2;
-            }
+        if (isRequirementAchieved(_transactionId) == true) {
+            confirmations[_transactionId].status = 2;
```

Comment:
Because the state is rolled back, as a technical matter, I don't think this makes a difference; but as a logical matter, it makes my brain wince that transaction status is first set to executed before the transaction is actually executed.

I think renaming from "Executed" to "Confirmed" in the enum and in the functions would resolve this peculiarity, and be more consistent with the mapping name "confirmations".

You could also add (back) an "Executed" status, but then you'd need to add in all of this code setting the status from confirmed to executed and I'm not sure that adds any real value.